app/src/main/java/ch/sdp/vibester/activity/DownloadActivity.kt

```diff
@@ -21,16 +21,23 @@ import ch.sdp.vibester.api.ItunesMusicApi
 import ch.sdp.vibester.model.Song
 import okhttp3.OkHttpClient
 import org.w3c.dom.Text
+import java.io.File
 import java.lang.IllegalArgumentException
 /*
  * Activity that handles downloading of song extracts.
  */
 class DownloadActivity : AppCompatActivity() {
+    //Companion object to indicate when the download completes.
+    companion object {
+        var downloadComplete = false
+    }
+
     private val STORAGE_PERMISSION_CODE = 1000
     private lateinit var song: Song
     private var songName: String = "imagine dragons believer"
     private var downloadId: Long = 0
```

Comment:
Is it hardcoded for now ? Or did you forget to remove the var use for debug ?

Reply:
Which one? All of them are being reused, if you meant the initial values it doesn't matter much. Can just change them around if required, since they will be overwritten.

Reply:
I was saying the songName ?

Reply:
I think the hardcoded songName is a placeholder testing value.